common/src/main/java/io/netty/util/NetUtil.java

```diff
@@ -850,18 +847,17 @@ public static Inet6Address getByName(CharSequence ip, boolean ipv4Mapped) {
 
         final boolean isCompressed = compressBegin > 0;
         // Finish up last set of data that was accumulated in the loop (or before the loop)
-        if (ipv4Seperators > 0) {
+        if (ipv4Separators > 0) {
             if (begin > 0 && i - begin > IPV4_MAX_CHAR_BETWEEN_SEPARATOR ||
-                    ipv4Seperators != IPV4_SEPARATORS ||
+                    ipv4Separators != IPV4_SEPARATORS ||
                     currentIndex >= bytes.length) {
                 return null;
             }
-            if (ipv6Seperators == 0) {
+            if (ipv6Separators == 0) {
                 compressLength = 12;
-            } else if (ipv6Seperators >= IPV6_MIN_SEPARATORS &&
-                           ip.charAt(ipLength - 1) != ':' &&
-                           (!isCompressed && (ipv6Seperators == 6 && ip.charAt(0) != ':') ||
-                            isCompressed && (ipv6Seperators + 1 < IPV6_MAX_SEPARATORS &&
+            } else if (ipv6Separators >= IPV6_MIN_SEPARATORS &&
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (7) used in the expression (maximum allowed 5). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1067)